test/e2e_env/kubernetes/meshfaultinjection/meshfaultijnection.go

```diff
@@ -0,0 +1,156 @@
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func MeshFaultInjection() {
+	mesh := "mfi-rules"
+	namespace := "mfi-rules-ns"
+	testServerURL := fmt.Sprintf("test-server.%s.svc:80", namespace)
+
+	meshIdentity := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshIdentity
+metadata:
+  name: identity
+  namespace: %s  
+  labels:
+    kuma.io/mesh: %s
+    kuma.io/origin: zone
+spec:
+  selector:
+    dataplane:
+      matchLabels: {}
+  spiffeID:
+    trustDomain: "{{ .Mesh }}.{{ .Zone }}.mesh.local"
+    path: "/ns/{{ .Namespace }}/sa/{{ .ServiceAccount }}"
+  provider:
+    type: Bundled
+    bundled:
+      meshTrustCreation: Enabled
+      insecureAllowSelfSigned: true
+      certificateParameters:
+        expiry: 24h
+      autogenerate:
+        enabled: true
+`, Config.KumaNamespace, mesh)
+
+	mtp := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  name: mtp-1
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+spec:
+  rules:
+    - default:
+        allow:
+          - spiffeID:
+              type: Prefix
+              value: spiffe://%s.default.mesh.local`, namespace, mesh, mesh)
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(Yaml(builders.Mesh().
+				WithName(mesh).
+				WithMeshServicesEnabled(mesh_proto.Mesh_MeshServices_Exclusive))).
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(YamlK8s(mtp)).
+			Install(Parallel(
+				democlient.Install(democlient.WithNamespace(namespace), democlient.WithMesh(mesh)),
+				testserver.Install(testserver.WithMesh(mesh), testserver.WithNamespace(namespace)),
+			)).
+			Install(YamlK8s(meshIdentity)).
+			Setup(kubernetes.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	E2EAfterAll(func() {
+		Expect(kubernetes.Cluster.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(kubernetes.Cluster.DeleteMesh(mesh)).To(Succeed())
+	})
+
+	E2EAfterEach(func() {
+		Expect(DeleteMeshResources(kubernetes.Cluster, mesh, v1alpha1.MeshFaultInjectionResourceTypeDescriptor)).To(Succeed())
+	})
+
+	DescribeTable("should fail based on spiffeID", func(mfiConfig string) {
+		// traffic should work normally
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectEchoResponse(
+				kubernetes.Cluster, "demo-client", testServerURL,
+				client.FromKubernetesPod(namespace, "demo-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.Instance).To(ContainSubstring("test-server"))
+		}, "30s", "1s").Should(Succeed())
+
+		// when
+		Expect(YamlK8s(mfiConfig)(kubernetes.Cluster)).To(Succeed())
+
+		// should fail on fault injection
+		Eventually(func(g Gomega) {
+			resp, err := client.CollectFailure(
+				kubernetes.Cluster, "demo-client", testServerURL,
+				client.FromKubernetesPod(namespace, "demo-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(resp.ResponseCode).To(Equal(500))
+		}, "30s", "1s").Should(Succeed())
```

Comment:
nitpick: maybe have 

```suggestion
		}, "30s", "1s").MustPassRepeatedly(3).Should(Succeed())
```

so that there's no false positives? 🤔

Reply:
changed